source/distributed/tria_base.cc

```diff
@@ -342,6 +351,196 @@ namespace parallel
 
 
 
+  template <int dim, int spacedim>
+  void
+  TriangulationBase<dim, spacedim>::reset_global_cell_indices()
+  {
+#ifndef DEAL_II_WITH_MPI
+    Assert(false, ExcNeedsMPI());
+#else
+
+    // currently only implemented for distributed triangulations
+    if (dynamic_cast<const parallel::DistributedTriangulationBase<dim, spacedim>
+                       *>(this) == nullptr)
+      return;
```

Comment:
That looks dangerous -- what happens for other triangulation? Do we get wrong information in those cases?